### ISSUE TITLE
Errors in afterAll functions are not reported.

### ISSUE THREAD

**Issue by johnjbarton** [bcaudan/jasmine-spec-reporter#210]
## Feature request 

**Current behavior**
Errors in afterAll() still allow tests to pass.
**Wanted behavior**
Tests fail on errors
**What is the motivation / use case for changing the behavior?**
Errors are bad ;-)

The suiteDone() can take a result argument:
https://github.com/bcaudan/jasmine-spec-reporter/blob/master/src/display/execution-display.ts#L101
jasmine will report afterAll errors as a failedExpectation in that arg.

We need that to 1) report the error 2) ensure the final status is FAILED.

**Comment by johnjbarton** [comment]
For test
```js
describe('afterall errors', () => {
	afterAll(() => {
		throw new Error('error in afterAll');
	});
	it('should not pass', () => {
		expect(true).toBe(true);
	});
});
```

Here is the default reporter from jasmine:
```shell 
./node_modules/jasmine/bin/jasmine.js afterall_throws_spec.js 
Started
.


1 spec, 1 failure
Finished in 0.01 seconds

An error was thrown in an afterAll
AfterAll Error: error in afterAll
```
Here is the jasmine-spec-reporter result:

```shell
[afterall]$ ./node_modules/jasmine/bin/jasmine.js afterall_throws_spec.js 
Jasmine started

  afterall errors
    ✓ should not pass

Executed 1 of 1 spec SUCCESS in 0.011 sec.
[afterall]$ 
```
I think this is a change in jasmine 2.5.

**Comment by bcaudan** [comment]
nice catch

**Comment by bcaudan** [comment]
Available in `jasmine-spec-reporter@4.2.0`

**Comment by johnjbarton** [comment]
I don't think the spec reporter should be exiting the process, generally
the testrunner should do that.

**Comment by bcaudan** [comment]
@johnjbarton is right, the reporter is not responsible for determining the result or exiting the process.

I think this issue is probably solved with jasmine@3.0.0.
Try to upgrade your jasmine version if you are on an older one, otherwise your issue should probably be reported at https://github.com/jasmine/jasmine.

**Comment by gvdp** [comment]
Yeah I agree, I was wondering if maybe the reporter "swallowed" the error somehow but afterwards I tried disabling the reporter and it produced the same behaviour. I was just wondering what you guys would do in such a situation. 
It's a protractor test and I found in the style guide that it's bad practice to do assertions/verifications in the afterAll() hook, so it seems to be by design that this wouldn't exit the process. For now I moved my assertions to the afterEach() hook.
